### PR TITLE
gup: 0.9.1 -> 0.9.2

### PR DESCRIPTION
**Pull request by timbertson**
## Description of changes

Fixes build error due to pylint changes (https://github.com/timbertson/gup/issues/23).

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/build-managers/gup/default.nix
Comment (smancill) [inline pkgs/development/tools/build-managers/gup/default.nix:8]:
```suggestion
    rev = "version-${version}";
```

I think we should keep it with `${version}`.

File: pkgs/development/tools/build-managers/gup/default.nix
Comment (smancill) [inline pkgs/development/tools/build-managers/gup/default.nix:9]:
```suggestion
    hash = "sha256-bV5HauM0xmRI/9Pxp1cYLPLA8PbFvPER2y4mAMmgchs=";
```

Could you use this opportunity to update to SRI hash?

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 336413` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gup</li>
  </ul>
</details>

**Review by smancill — APPROVED** [review]

**Comment by timbertson** [comment]
Thanks for the review @smancill , updated as requested (and updated the _update_ script so it won't keep clobbering the `version` attr 🤦 ).

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 336413` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gup</li>
  </ul>
</details>

**Review by smancill — APPROVED** [review]
LGTM.

And you are the owner of `nix-update-source` so I'll take it you know what it's needed there.

File: pkgs/development/tools/build-managers/gup/default.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/gup/default.nix:9]: hash should stay at this line

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Why is pylint run ad an linter even?! Please remove that, we don't really care if any lint fails.

File: pkgs/development/tools/build-managers/gup/default.nix
Comment (timbertson) [reply pkgs/development/tools/build-managers/gup/default.nix:9]: This moved because I updated to an [SRI hash](https://github.com/NixOS/nixpkgs/pull/336413#discussion_r1740267651), and the update script orders attributes alphabetically (`sha256` -> `hash`). It won't need to change ordering again in the future.

**Comment by timbertson** [comment]
Thanks for the review @SuperSandro2000 

> Why is pylint run ad an linter even?! Please remove that, we don't really care if any lint fails.

It's not linting the source code as you'd typically use it, it's checking that [the custom build process](https://github.com/timbertson/gup/blob/3ce059fdbd6cc3da2cbcb476da4ae61a6fb32153/python/build/combine_modules.py) didn't introduce any obvious errors when merging multiple python modules into one file, like undefined functions or naming clashes. It used to use pychecker, but that is now unmaintained.

**Comment by SuperSandro2000** [comment]
Still und package could break with every linter update and we generally want to be able to merge updates to them without causing breakages.

**Comment by timbertson** [comment]
> the package could break with every linter update

If I (as package maintainer) accept that risk, can we proceed? This is fundamentally a version bump pr, using pylint in the derivation is not optional unless you want to patch upstream to remove its use.

I understand that you don't think this is a good idea, but if you really care can we have a proper discussion about potential solutions and their tradeoffs on [the repo itself](https://github.com/timbertson/gup), rather than holding up this PR?

**Comment by timbertson** [comment]
@SuperSandro2000 you're currently blocking this PR. Can you please respond to my above comment, and clearly state your requirements for unblocking? It sounds like current status is this PR will be blocked (leaving the package broken) until I make changes to remove pylint from the build process. But I want to confirm that's still the case after my last comment.

**Comment by timbertson** [comment]
Note also that pylint is invoked with `--errors-only`, it's not going to fail on typical lint warnings.

The issue that needed fixing here was a (possible) use-before-assignment of `xrange` due to some py2/3 compat logic, which has now been removed.

File: pkgs/development/tools/build-managers/gup/default.nix
Comment (timbertson) [reply pkgs/development/tools/build-managers/gup/default.nix:9]: It's a one-time change, that doesn't seem worth migrating tools for.

Comment (SuperSandro2000) [reply pkgs/development/tools/build-managers/gup/default.nix:9]: That update script are just using 4 packages or so. Maybe we should just drop it.

**Review by SuperSandro2000 — APPROVED** [review]
yeah alright

**Pull request merged**
